framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C

```diff
@@ -1028,6 +1028,10 @@ DistributedRectilinearMeshGenerator::generate()
   mesh->set_mesh_dimension(_dim);
   mesh->set_spatial_dimension(_dim);
 
+  // Let the mesh know that it's not serialized; the libMesh idiom
+  // here "deletes" 0 elements.
+  mesh->delete_remote_elements();
+
```

Comment:
`auto mesh = _mesh->buildMeshBaseObject(MooseMesh::ParallelType::DISTRIBUTED);`

Will libmesh figure out we are working on a distributed mesh?

Reply:
Well that's the trick.  libMesh definitely knows that you're trying to build a DistributedMesh, but not that you're trying to start with a *distributed* mesh, as opposed to a temporarily-serialized mesh which you merely want the capability to distribute later.  Only when you finally tell it to get rid of any remote elements does DistributedMesh decide "I'm definitely not serialized anymore".

I could be convinced to change the default behavior, I think, since it was originally sort of for backwards compatibility purposes that are now moot, but for now making the serialization status explicit in user code is the only fix (and even in the future it'll be the most robust thing to do).

Thanks for looking at this so promptly.

Reply:
I see.

Reply:
Since no nodes or elements have been added, wouldn't simply calling `mesh->set_distributed()` be sufficient here?

Reply:
It should be, I think.  Not sure why I used the old idiom other than force of habit.